### PR TITLE
Dont run with scissors

### PR DESCRIPTION
**Pull request by katekerllenevich**
Ported from this mod by @JustDoom: https://www.curseforge.com/minecraft/mc-mods/dont-run-with-scissors

If you sprint while holding scissors, you will take constant damage.

### CODE REVIEW

File: patches/server/0173-Dont-run-with-scissors.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
+index 9f93e2f5aed24ad0c13076fc6b2087d988bd0f01..35a9f1bf592adc96674a9bf5324373671c219ffc 100644
+--- a/src/main/java/net/minecraft/server/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -1388,6 +1388,12 @@ public class PlayerConnection implements PacketListenerPlayIn {
+                                     this.player.fallDistance = 0.0F;
+                                 }
+ 
++                                // Purpur Start
++                                if (this.player.isSprinting() && this.player.getWorldServer().purpurConfig.dontRunWithScissors && this.player.getBukkitEntity().getInventory().getItemInMainHand().getType().equals(org.bukkit.Material.SHEARS)) {
```
Comment (BillyGalbreath) [inline patches/server/0173-Dont-run-with-scissors.patch]: should we check offhand, too? its just as dangerous, if not more, than the main hand

**Review by BillyGalbreath — APPROVED** [review]

File: patches/server/0173-Dont-run-with-scissors.patch (hunk 1 above)
Comment (katekerllenevich) [reply patches/server/0173-Dont-run-with-scissors.patch]: yeah

Comment (katekerllenevich) [reply patches/server/0173-Dont-run-with-scissors.patch]: fairly certain i did it correctly

**Comment by katekerllenevich** [comment]
something i do realize looking at the mod, is that the damage only happens sometimes.
is that something i should implement?

**Comment by katekerllenevich** [comment]
Adding it makes it feel better. Its now ready for merge.

**Comment by JustDoom** [comment]
i did not expect for this to be added to purpur xD

**Pull request merged**